src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -90,6 +90,28 @@ static function (ReflectionProperty $property) : bool {
         ));
     }
 
+    /**
+     * @return list<string>
+     */
+    private function generatePropertyHydrateCall(ObjectProperty $property, string $input) : array
+    {
+        $ret = [];
+
+        $propertyName = $property->name;
+        $escapedName = var_export($propertyName, true);
+
+        if ($property->type && !$property->required && !$property->hasDefault) {
+            $ret[] = "\$object->{$propertyName} = {$input}[{$escapedName}] ?? null;";
+        } else {
+            $ret[] = "if (isset({$input}[{$escapedName}]) || \$object->{$propertyName} !== null "
```

Comment:
This causes a property read on a potentially un-initialized property: unless I'm misreading this, it will lead to a `TypeError`

Reply:
That is the tricky part: if property has no type set, or if it has a default value, then direct access is OK, whichever was the instantiation mechanism (`newInstanceWithoutConstructor()` or simple `new Foo` both will initialize properly the default values).

For the use case where the property type is not optional (does not allows null) this code can lead to a `TypeError`, if it doesn't, then it will leave an partially uninitialized object. That's the only use case where it may happen.

I don't know how this API should deal with this, should it let live the partially uninitialized object, or should it raise a more explicit error or should it just let the `TypeError` pass ?

Reply:
There's a way by checking `ReflectionProperty#isInitialized()`, but that has indeed a performance impact :thinking:

Reply:
Actually, it is also possible to verify if the `(array)` cast version of the object contains the key in question

Reply:
It seems that `testHydratorWillNotRaisedUnitiliazedTypedPropertyAccessError()` tests the following use cases:

- optional typed property has a default, and property is not set in the array,
- required typed property has a default, and property is not set in the array,
- optional typed property has no default, and property is not set in the array, is set to null,
- required typed property has no default, and property has a value in the array.

The only use case it doesn't test is:

- required typed property has no default, and property is not set in the array, which should lead to the error.

Reply:
> There's a way by checking ReflectionProperty#isInitialized(), but that has indeed a performance impact thinking

You should not use reflection at runtime (even thought it's very fast with PHP) - that's the idea of using reflection at generation time to write alternative code whenever necessary. The question is: which code may we write to handle this case ? The only way to know is to answer the "what should we do in this case, be resilient and leave a broken object, or fail ?" question.

Reply:
yeah, as per discussion in the issue (factories being "complete") I think we're OK with assuming that all fields are included in the given `$data`

Reply:
So I assume there is nothing to be done here ?

Reply:
Nothing to be done here :+1: